synapse/python_dependencies.py

```diff
@@ -116,6 +113,12 @@
     if name not in ["systemd", "lint"]:
         ALL_OPTIONAL_REQUIREMENTS = set(optional_deps) | ALL_OPTIONAL_REQUIREMENTS
 
+# We exclude lint and test as those are developer requirements
+ALL_RUNTIME_REQUIREMENT_COMPONENTS = set(CONDITIONAL_REQUIREMENTS.keys()) - {
+    "lint",
+    "test",
+}
+
```

Comment:
@richvdh This change filters out all runtime dependencies from all conditional dependencies, by only excluding `lint` and `test`, and then flattening the resulting dependencies into a single `set`.

I could change this to be only a set with `{"lint", "test"}`, as this piece is only used in one part later on.